upload/enqueue.go

```diff
@@ -52,15 +53,23 @@ func (job *EnqueuedJob) ID() string {
 }
 
 // albumID returns the album ID of the created (or existent) album in PhotosService.
+// It uses cache to reduce number of request to PhotoService.
 func (job *EnqueuedJob) albumID() (string, error) {
 	// Return if empty to avoid a PhotosService call.
 	if job.AlbumName == "" {
 		return "", nil
 	}
 
+	albumID, err := job.AlbumCache.Get(job.AlbumName)
```

Comment:
There's a race here - two workers can call Get on an empty cache and then proceed to both call `GetOrCreateAlbumByName`.

We think that this can lead to both of those workers generating new albums, because we think the API's `list` call is eventually consistent. Let's say that happens, and the two calls to `GetOrCreateAlbumByName` both create an album, and return two different album ids.

Then, our two workers will then race to put different values into the album cache on L73, and _one_ of them puts the photo they're working on into the "wrong" album.

Here's a timeline:

Worker one checks cache for album "test album", finds it is empty
Worker two checks cache for album "test album", finds it is empty
Worker one gets album id 10
Worker two gets album id 11
Worker one puts "test album" -> 10 into the cache
Worker two puts "test album" -> 11 into the cache
Worker one uploads a photo to album 10
Worker two uploads a photo to album 11
...and every other photo in this album ends up in album 11

If we want to use the Cache approach, we need to add something like `ComputeIfAbsent` to it; even if the Get/Put/Invalidate methods are thread safe, it is still possible to use them in an unsound way :-)